xls/dslx/stdlib/abs_diff.x

```diff
@@ -0,0 +1,194 @@
+// Copyright 2025 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Functionality for computing an absolute difference between two numbers with a
+//! "residual" correction term. By keeping a correction term we avoid eagerly
+//! incorporating a second carry chain in computing the final result.
+//!
+//! Sample usage:
+//!
+//! ```dslx
+//! let x: u4 = u4:0;
+//! let y: u4 = u4:1;
+//! let adr: AbsDiffResult<4> = abs_diff(x, y);
+//! assert_eq(to_corrected(adr), u4:1);
+//! assert_eq(is_x_larger(adr), false);
+//! assert_eq(is_zero(adr), false);
+//! ```
+
+import std;
+
+/// Describes the result of an absolute difference operation.
+///
+/// By keeping the result in two terms with one being a correction term, we
+/// can avoid a second carry chain being applied in the operation.
+///
+/// Additionally, this information can be used to tell us which of the two operands
+/// was larger -- use `is_x_larger` for this.
+pub struct AbsDiffResult<N: u32> {
+    correction: bool,
+    /// Note: this is the *uncorrected* absolute difference -- the `correction` term
+    /// must be added to get the true absolute difference -- use `to_corrected` to
+    /// flatten this structure to the absolute difference value.
+    uncorrected: uN[N],
+}
+
+/// Returns `true` if the first operand was larger, `false` if the second operand was larger.
+pub fn is_x_larger<N: u32>(adr: AbsDiffResult<N>) -> bool { adr.correction }
```

Comment:
nit: what about `has_correction` or `is_positive` ? I feel like `x` here would not really be visible to the reader: 
- it's not really a member of the `AbsDiffResult`.
- being a "positional" argument of `abs_diff` the name `x` is only known to "people who read the doc".